pennylane_lightning/core/src/simulators/lightning_gpu/StateVectorCudaMPI.hpp

```diff
@@ -401,9 +401,11 @@ class StateVectorCudaMPI final
         } else if (opName == "Rot" || opName == "CRot") {
             auto rot_matrix =
                 adjoint
-                    ? cuGates::getRot<CFP_t>(params[2], params[1], params[0])
+                    ? cuGates::getRot<CFP_t>(-params[2], -params[1], -params[0])
                     : cuGates::getRot<CFP_t>(params[0], params[1], params[2]);
-            applyDeviceMatrixGate(rot_matrix.data(), ctrls, tgts, adjoint);
+            // Use default adjoint=false for applyDeviceMatrixGate() as reversed
+            // rot_matrix explicitly used
+            applyDeviceMatrixGate(rot_matrix.data(), ctrls, tgts);
```

Comment:
I'm afraid this is not yet clear why we had to prepare the adjoint matrix for `Rot` manually! This issue can be re-produced for any multi-param gates. `custatevecApplyMatrixGetWorkspaceSize` and `custatevecApplyMatrix` API calls were supposed to take care of these sign updates, right? :thinking: cc: @multiphaseCFD

Reply:
Indeed `custatevecApplyMatrixGetWorkspaceSize` and `custatevecApplyMatrix` can take care of the adjoint signs. I assumed supplying the adjoint matrix manually would give better performance. I have just tested both methods (supplying the adjoint matrix vs using the adjoint from the custatevec API), and there's no performance difference. I can change it to let custatevec deal with the adjoint now.

Reply:
There you go, you got it right! thanks 🙌